app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/parser/OpeningHoursParser.kt

```diff
@@ -53,9 +53,30 @@ fun OpeningHoursRuleList.toOpeningHoursRows(): List<OpeningHoursRow>? {
     return result
 }
 
+/** returns null if the list of rules cannot be displayed by the opening hours widget */
+fun OpeningHoursRuleList.toCollectionTimesRows(): List<OpeningHoursRow>? {
+    if (!isSupportedCollectionTimes()) {
+        // parsable, but not supported by StreetComplete
+        return null
+    }
+
+    val result = mutableListOf<OpeningHoursRow>()
+
+    for (rule in rules) {
+        if (rule.modifier != null && rule.modifier!!.isSimpleOff()) {
+            result.add(rule.createOffDays())
+        } else {
+            result.addAll(rule.createOpeningWeekdays())
+        }
+    }
+
+    return result
+}
+
 /* ---------------------------------- Checks if it is supported --------------------------------- */
 
 fun OpeningHoursRuleList.isSupported(): Boolean = rules.isSupported()
+fun OpeningHoursRuleList.isSupportedCollectionTimes(): Boolean = rules.isSupportedCollectionTimes()
```

Comment:
There is some code duplication now. It might be cleaner to have a Boolean parameter in the various isSupported functions to handle the differences?

Reply:
No, that's okay. It is for two different UIs which also do not inherit anything from each other.

Except the `OpeningHoursRow` data structure, about which I am not happy about anyway.